### PR TITLE
Feature/s3 module

### PR DESCRIPTION
**Pull request by chriswynnyk**
Template to create a "transfer" s3 bucket for RCCL.

ACL's for external users would be handled outside of terraform.

### CODE REVIEW

File: transfer/policy.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "*"
+      },
+      "Resource": "arn:aws:s3:::${bucket}/*",
+      "Sid": "transfer-bucket-policy"
+    }
+  ],
+  "Version": "2012-10-17"
+}
```
Comment (isaacanthony) [inline transfer/policy.json]: `\n`

File: transfer/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+data "template_file" "policy" {
+  template = "${file("${path.module}/policy.json")}"
+
+  vars = {
+    bucket     = "${var.name}-${var.environment}-transfer"
+    account_id = "${var.account_id}"
+  }
+}
+
+resource "aws_s3_bucket" "transfer" {
```
Comment (njgerner) [inline transfer/main.tf]: you guys are going to want to have more control over the object versioning, cloudwatch logging and server side encryption

Versioning:
https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#using-versioning

Logging:
https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#enable-logging

Encryption:
https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#enable-default-server-side-encryption

File: transfer/policy.json (hunk 1 above)
Comment (njgerner) [reply transfer/policy.json]: this doesn't seem right, seems to give any AWS account access to any object

File: transfer/main.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
+variable "name" {
+}
+
+variable "environment" {
+}
+
+variable "account_id" {
```
Comment (isaacanthony) [inline transfer/main.tf]: This variable is no longer needed.

File: s3/main.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  deletion_window_in_days = 10
+}
+
+resource "aws_s3_bucket" "s3" {
+  bucket = "${var.name}-${var.environment}"
+  acl    = "private"
+
+  logging {
+    target_bucket = "${var.log_bucket}"
+    target_prefix = "log/${var.name}/"
```
Comment (njgerner) [inline s3/main.tf]: Will want environment here too

**Pull request merged**